packages/opentelemetry-plugin-mongodb/src/mongodb.ts

```diff
@@ -0,0 +1,208 @@
+/*!
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// mongodb.Server type is deprecated so every use trigger a lint error
+/* tslint:disable:deprecation */
+
+import { BasePlugin } from '@opentelemetry/core';
+import { Span, SpanKind, CanonicalCode } from '@opentelemetry/types';
+import * as mongodb from 'mongodb';
+import * as shimmer from 'shimmer';
+
+type Func<T> = (...args: unknown[]) => T;
+type MongoDB = typeof mongodb;
+interface MongoInternalCommand {
+  findandmodify: boolean;
+  createIndexes: boolean;
+  count: boolean;
+  ismaster: boolean;
+}
+
+/** MongoDB instrumentation plugin for OpenTelemetry */
+export class MongoDBPlugin extends BasePlugin<MongoDB> {
+  private readonly _SERVER_METHODS = ['insert', 'update', 'remove', 'command'];
+  private readonly _CURSOR_METHODS = ['_next', 'next'];
+
+  protected readonly _supportedVersions = ['>=2 <3'];
```

Comment:
should use `supportedVersions` from `BasePlugin` and add tests in order to be sure that plugin is not loaded for other versions

Reply:
I changed it to use `supportedVersions` but i don't think we need to test that it behave correctly in the plugin, it would argue that it need to be done only where the behavior is implemented (ie. in PluginLoader)